Automattic/wp-calypso
# Site Settings: Refactor Monitor settings to `react-query` and remove UNSAFE methods

#### Changes proposed in this Pull Request

This PR migrates the site monitor settings to use `react-query`, as their usage is very simple and doesn't need all the Redux boilerplate. We also clean up the unused site monitor state and query component (and their tests).

We also refactor away from the `UNSAFE_` deprecated React component methods in the monitor settings form. That's easy with react-query, because we no longer need to manage any component state.

Part of #58453, where it addresses 2 items.

#### Testing instructions
* Go to `/settings/security/:site` where `:site` is a Jetpack site.
* Play with enabling and disabling the main toggle under "Downtime monitoring".
* Verify that all toggles still end up disabled while activating / deactivating the Monitor module.
* Verify that the settings toggles (subtoggles) are inactive when the module (main toggle) is inactive.
* Play with activating and deactivating each setting subtoggle in "Downtime monitoring".
* Verify that while updating a setting, all controls still get disabled.
* Verify that the settings still save properly, and are optimistically updated.
* Verify that you get the same success and error notices.
* Disable the corresponding network request and verify that on error, the settings are reverted to their previous values.
* Verify all tests still pass.
